src/Stryker.Core/Stryker.Core/Compiling/RollbackProcess.cs

```diff
@@ -94,21 +94,16 @@ public RollbackProcessResult Start(CSharpCompilation compiler, ImmutableArray<Di
 
         private int? ExtractMutationIfAndId(SyntaxNode node)
         {
-            var first = node.GetAnnotations(MutantPlacer.MutationMarkers).FirstOrDefault();
-            if (first == null)
+            var (engine, id) = MutantPlacer.FindEngine(node);
+
+            if (engine == null)
             {
                 return null;
             }
 
-            if (first.Data == null)
-            {
-                Logger.LogDebug("Remove a mutation helper.");
-                return _dummyId;
-            }
-            var mutantId = int.Parse(first.Data);
-            Logger.LogDebug("Found id {0} in MutantIf annotation", mutantId);
-            return mutantId;
+            Logger.LogDebug(id == -1 ? $"Found a helper: {engine}." : $"Found id {id} in {engine} annotation.");
```

Comment:
```suggestion
            Logger.LogDebug(id == -1 ? $"Found a mutation engine: {engine}." : $"Found id {id} in {engine} annotation.");
```

Reply:
I am not confortable with your proposal, as `engine` is not a _mutation_ engine, but more of an `injection`engine. Just to clarify any potential misunderstanding:
- if `id` is positive or null, this is a mutation, and the engine will describe the control logic: `IfInstrumentationEngine` (`if (Stryker.MutantIsActive(1)){...}else{...}`) or `ConditionalInstrumentationEngine` (`Stryker.MutantIsActive(1)?...:...`).
- if ' `id` is `-1` (magic number ☹️), this is some other helping construction, such `return default(...)` or an expression to body conversion, or out parameters initialization.... Definitely not a **mutation**, more a minor refactoring that helps mutation to succeed

Reply:
I'm fine with removing mutation from the logrule. I mainly meant to place engine instead of helper because we definitely know it's an engine.

Reply:
well that's an engine that injects and remove helpers :-)

Reply:
Anyway, I improved the error message on the 'master' PR, assuming it will be the one you pick. I also fixed another mistake: https://github.com/stryker-mutator/stryker-net/pull/1529/files#diff-edba80c123cba4b1cd940aeec21549583d3abeb379d33e2ea3545516649925baR127